tiledb/api/src/array/mod.rs

```diff
@@ -380,6 +382,113 @@ impl Array {
         Ok(Some(datatype))
     }
 
+    pub fn vacuum<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_vacuum(ctx, c_array_uri.as_ptr(), config.capi())
+        })?;
+        Ok(())
+    }
+
+    pub fn upgrade_version<S>(
```

Comment:
I was considering using a lambda to simplify the wrapping if possible, since the code is very repetitive. The lambda would be either tiledb_array_vaccum, tiledb_upgrade_version, or another call with three arguments {context, ari, config}

Reply:
Either way is fine with me.

Reply:
left as is